app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

```diff
@@ -159,51 +182,85 @@ public void speechResultCallback(ArrayList voiceResult) {
         new RedditSearcher(this, firstResult, getActivity().getApplicationContext()).getRedditComment();
     }
 
-    @Override
-    public void partialCallback(ArrayList partialResult) {
+    @Override public void partialCallback(ArrayList partialResult) {
         speechTextDisplay.setText(partialResult.get(0).toString());
     }
 
-    @Override
-    public void errorCallback(int errorCode, int numErrors) {
+    @Override public void errorCallback(int errorCode, int numErrors) {
         isListening = false;
         updateListeningIndicator();
         Log.d(DEBUG_TAG, "Got error, stopped listening.");
 
-        if (numErrors == 1) { // to prevent showing multiple toasts
+        if (numErrors == 1) { // to prevent repeating errors
             if (errorCode == SpeechRecognizer.ERROR_NO_MATCH) { // error 7
                 //TODO: change this to saying out loud, "please try again"
                 Toast.makeText(getActivity().getApplicationContext(),
                         "Error: Speech was not recognized.", Toast.LENGTH_SHORT).show();
+                showComment("make sense pls");
             } else if (errorCode == SpeechRecognizer.ERROR_SPEECH_TIMEOUT) { //error 6
                 Toast.makeText(getActivity().getApplicationContext(),
                         "Error: Please say something.", Toast.LENGTH_SHORT).show();
+                showComment("you have a mouth right");
             } else {
                 Toast.makeText(getActivity().getApplicationContext(),
                         "Error occurred! Try again.", Toast.LENGTH_SHORT).show();
+                showComment("error on our side, sorry :'-(");
             }
         }
     }
 
-    @Override
-    public void commentCallback(String comment) {
+    @Override public void commentCallback(String comment) {
         if (comment == null) {
             Log.d(DEBUG_TAG, "No valid comments found");
             Toast.makeText(getContext(), "No valid comments available", Toast.LENGTH_SHORT).show();
+            showComment("No valid comments found");
         } else {
-            commentText.setText(comment);
-            mListener.speak(comment);
+            showComment(comment);
         }
     }
 
+    public void showComment(String comment){ //a comment will always show, so do it here
```

Comment:
I think it is because the master has not been merged to that branch. If there is no voice input, the app will will crash because voiceInput.get(0) is null